integration-tests/src/test/kotlin/com/google/devtools/ksp/test/PlaygroundIT.kt

```diff
@@ -65,6 +65,28 @@ class PlaygroundIT {
         project.restore("workload/build.gradle.kts")
     }
 
+    @Test
+    fun testArbitraryBuildDir() {
+        Assume.assumeTrue(System.getProperty("os.name").startsWith("Windows", ignoreCase = true))
+        val gradleRunner = GradleRunner.create().withProjectDir(project.root)
+
+        File(project.root, "workload/build.gradle.kts")
+            .appendText("project.buildDir = File(\"D:/build/\")")
```

Comment:
note that it does seem like there is only one C drive on github windows runners, therefore this test might not be testable on github runners.